packages/chain-controller/src/SnapHandlerClient.test.ts

```diff
@@ -0,0 +1,72 @@
+import type { SnapController } from '@metamask/snaps-controllers';
+import type { SnapId } from '@metamask/snaps-sdk';
+import { HandlerType } from '@metamask/snaps-utils';
+
+import { SnapHandlerClient } from './SnapHandlerClient';
+
+describe('SnapHandlerClient', () => {
+  const handleRequest = jest.fn();
+
+  const snapId = 'local:localhost:3000' as SnapId;
+  const snapController = {
```

Comment:
Similar as previous — do you need the concept of a SnapController here? Maybe you just need to pass the `handleRequest` around.